adapter/src/services/adapterService.ts

```diff
@@ -0,0 +1,45 @@
+import { Interpreter } from "../interpreter/Interpreter";
+import { AdapterConfig } from "../model/AdapterConfig";
+import { DataImportResponse } from "../model/DataImportResponse";
+
+import { ProtocolConfig } from "../model/ProtocolConfig";
+
+export class AdapterService {
+    /**
+     * @description Create an instance of AdapterService
+     */
+    constructor () {
+    }
+
+    // To Implement
+    static getAllFormats(): Array<Interpreter> {
+      try {
+        // TODO implement interpreter
```

Comment:
You can simplify the old very generic implementation by just listing all interpreters / importers I think...